src/components/OffCanvas/OffCanvasBody.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import * as baseTheme from './OffCanvas.scss';
+import {classNames} from '@shopify/react-utilities/styles';
+
+export interface State{
+    isMenuOpened:false
+}
+export interface Props{
+    width?:number,
+    transitionDuration?:number,
+    isMenuOpened?:boolean,
+    position?:any,    
+    children?:React.ReactNode,    
+    style?:any,
+    animation?:string
+}
+
+export class OffCanvasBody extends React.PureComponent<Props,State>
+{
+  state: State = { isMenuOpened:false };
+  render(){
+    const{
+      width = 270,
+      transitionDuration = 270,
+      isMenuOpened = false,
+      position = 'right',
+      children,      
+      style,
+      animation,      
+    } = this.props;
+
+    const className = classNames(
+      baseTheme.bodyClass  
+    );    
+    
+    let translateX = position === 'left' ? 0 : 0;
+    let closedStyle = {      
+      transitionDuration: transitionDuration + 'ms',
+      transform: 'translate(' + translateX + 'px, 0px)',
+      backfaceVisibility: 'hidden'
+    };
+    
+    let translateOpenX = position === 'left' ? width : 1 * width;
+    let openStyle = {
+      transform: 'translate(' + translateOpenX + 'px, 0px)'
+    };
+        
+    let currStyle = Object.assign({}, closedStyle);
+    if (isMenuOpened) {
+      currStyle = Object.assign({}, currStyle, openStyle);
+    }    
+    var finalStyle;
```

Comment:
Make finalStyle const and use ternary operator to set.  Also use === for comparison.  Ex: const finalStyle = animation === ‘Reveal’ ? currStyle : style;.